src/routes/communities/index.js

```diff
@@ -0,0 +1,212 @@
+import Joi from '../../joi';
+import models from '../../models';
+
+const createCommunitySchema = {
+  body: Joi.object({
+    organizers: Joi.number()
+      .required(),
+    members: Joi.number()
+      .required(),
+    name: Joi.string()
+      .max(50)
+      .required(),
+    content_types: Joi.string()
+      .required(),
+    description: Joi.string()
+      .max(250)
+      .required(),
+    image_path: Joi.string(),
+    tags: Joi.string(),
+    website: Joi.string(),
+    rules: Joi.string(),
+  }),
+};
+const updateCommunitySchema = {
+  body: Joi.object({
+    members: Joi.number(),
+    name: Joi.string()
+      .max(50),
+    content_types: Joi.string(),
+    description: Joi.string()
+      .max(250),
+    image_path: Joi.string(),
+    tags: Joi.string(),
+    website: Joi.string(),
+    rules: Joi.string(),
+  }),
+};
+
+const create = async (req, res) => {
+  const { error } = createCommunitySchema.body.validate(req.body);
+  if (error) {
+    return res.status(400).send({
+      errors: error.details,
+    });
+  }
+  const {
+    members, name, content_types, description, image_path, tags, website, rules,
+  } = req.body;
+  const image = await models.images.create({
+    user_id: req.user.id,
+    name: image_path,
+    path: image_path,
+  });
+  const community = await models.communities.create({
+    organizers: req.user.id,
+    members,
+    name,
+    content_types,
+    description,
+    image_id: image.id,
+    tags,
+    website,
+    rules,
+  });
+  res.send({
+    community,
+  });
+};
+
+const detail = async (req, res) => {
+  const { id } = req.params;
+  try {
+    const community = await models.communities.findOne({
+      where: {
+        id,
+      },
+      include: [
+        {
+          model: models.images,
+          as: 'image',
+        },
+      ],
+    });
+
+    if (!community) {
+      return res.send({
+        errors: [
+          {
+            message: 'Communities not found or you don\'t have a permission!',
+          },
+        ],
+      });
+    }
+    return res.send(community);
+  } catch (err) {
+    return res.status(500).send({
+      errors: [
+        {
+          message: err.message,
+        },
+      ],
+    });
+  }
+};
+
+const getCommunities = async (req, res) => {
+  const { limit } = req.query;
+  const community = await models.communities.findAll({
+    limit,
+    include: {
+      model: models.images,
+      as: 'image',
+    },
+  });
+  return res.send({ community, count: community.length });
+};
+
+const update = async (req, res) => {
+  const { error } = updateCommunitySchema.body.validate(req.body);
+  if (error) {
+    return res.status(400).send({
+      errors: error.details,
+    });
+  }
+  const { id } = req.params;
+  try {
+    const community = await models.communities.findOne({
+      where: {
+        id,
```

Comment:
Burada güvenlik açığı olmaması için isteği atan kullanıcının ozganizatör olup olmadığı da kontrol edilmeli.